lib/tree.rb

```diff
@@ -1,66 +1,135 @@
 class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
-
-   def initialize(key, val)
+  
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
   def initialize
     @root = nil
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  # Time Complexity: O(n) worst case
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
👍 , Since you have a recursive helper method you have a space complexity of O(log n) if the tree is balanced or O(n) if not.